docs/testing_plan.md

```diff
@@ -4,27 +4,123 @@
+Our goal for testing our server side code was to achieve the 50% coverage on our controller classes. We also wanted to ensure that the tests were ran in the pipeline and the coverage requirements were met before being able to merge a pull request. Example tests can be found under `server/tests/unit`. We have also added a few integration tests for the UserController under `server/tests/integration` which will be discussed further down in this document.
 
 ### -- Libraries Used --
 
  - [Mocha](https://mochajs.org/) (test framework)
  - [Sinon](https://sinonjs.org/) (Mocking library)
  - [Chai](https://www.chaijs.com/) (Assertion Library)
  - [Istanbul](https://istanbul.js.org/) (Coverage Library)
+ - [Supertest](https://github.com/visionmedia/supertest) (API Integration test library)
 
 #### Why we used these libraries:
 
-To test our controllers which are written in Typescript, we selected [Mocha](https://mochajs.org/) as our test framework. Mocha is open-source, extremely popular and offers extensive documentation and community support which made setting up our test environment easy. It also makes asynchronous testing simple which is a necessity for our application. We then selected [Sinon](https://sinonjs.org/) and [Chai](https://www.chaijs.com/) for all our mocking and assertion needs respectively. Again these are both open-source massively popular frameworks that are used in industry for stubs and assertions. Lastly we went with [Istanbul](https://istanbul.js.org/) which is a code coverage library that integrates with our testing framework, Mocha. It is highly configurable and can be used to output coverage reports or to block deployments which do not meet coverage requirements.
+To test our controllers which are written in Typescript, we selected [Mocha](https://mochajs.org/) as our test framework. Mocha is open-source, extremely popular and offers extensive documentation and community support which made setting up our test environment easy. It also makes asynchronous testing simple which is a necessity for our application. We then selected [Sinon](https://sinonjs.org/) and [Chai](https://www.chaijs.com/) for all our mocking and assertion needs respectively. Again these are both open-source massively popular frameworks that are used in industry for stubs and assertions. Lastly we went with [Istanbul](https://istanbul.js.org/) which is a code coverage library that integrates with our testing framework, Mocha. It is highly configurable and can be used to output coverage reports or to block deployments which do not meet coverage requirements. We decided to use supertest as it makes integration testing our api easy by simply creating a new instance of our application and passing it to the library to make HTTP requests and assertions.
```

Comment:
```suggestion
To test our controllers which are written in Typescript, we selected [Mocha](https://mochajs.org/) as our test framework. Mocha is open-source, extremely popular and offers extensive documentation and community support which made setting up our test environment easy. It also makes asynchronous testing simple which is a necessity for our application. We then selected [Sinon](https://sinonjs.org/) and [Chai](https://www.chaijs.com/) for all our mocking and assertion needs respectively. Again these are both open-source massively popular frameworks that are used in industry for stubs and assertions. Lastly we went with [Istanbul](https://istanbul.js.org/) which is a code coverage library that integrates with our testing framework, Mocha. It is highly configurable and can be used to output coverage reports or to block deployments which do not meet coverage requirements. We decided to use [Supertest](https://github.com/visionmedia/supertest) as it makes integration testing our api easy by simply creating a new instance of our application and passing it to the library to make HTTP requests and assertions.
```
Just making Supertest clickable to be consistent with the rest of your paragraph.